lib/ruby_lsp/requests/document_link.rb

```diff
@@ -114,6 +186,24 @@ def resolve_version(uri)
 
         GEM_TO_VERSION_MAP[uri.gem_name]
       end
+
+      sig { params(node: SyntaxTree::Node, name: String).void }
+      def add_rails_document_link(node, name)
+        # ignore non-Ruby files
+        return unless @uri.match?(/\.rb$/)
+
+        candidate = self.class.rails_documents.bsearch { |ary| ary[0] <=> name }
```

Comment:
To avoid having to use a binary search on every node that matches, can we switch the implementation to using a hash where the key is the thing we want to find documentation for?

I mean pre-building and then doing straight access on something like
```ruby
@rails_documents = {
  "belongs_to" => "url_for_belongs_to",
  "ActionController::Base" => "url_for_action_controller_base"
}
```